diff --git a/Makefile b/Makefile
index c4bbbade..cd0ed6db 100644
--- a/Makefile
+++ b/Makefile
@@ -42,7 +42,7 @@ POSTGRES_IMAGE_NAME ?= $(shell grep 'DefaultImageName.*=' "pkg/versions/versions
 KUSTOMIZE_VERSION ?= v5.3.0
 KIND_CLUSTER_NAME ?= pg
 KIND_CLUSTER_VERSION ?= v1.28.0
-CONTROLLER_TOOLS_VERSION ?= v0.13.0
+CONTROLLER_TOOLS_VERSION ?= v0.14.0
 GORELEASER_VERSION ?= v1.22.1
 SPELLCHECK_VERSION ?= 0.35.0
 WOKE_VERSION ?= 0.19.0
diff --git a/internal/cnpi/plugin/client/lifecycle.go b/internal/cnpi/plugin/client/lifecycle.go
index 00455ecd..6a564234 100644
--- a/internal/cnpi/plugin/client/lifecycle.go
+++ b/internal/cnpi/plugin/client/lifecycle.go
@@ -125,7 +125,13 @@ func (data *data) LifecycleHook(
 			continue
 		}
 
-		responseObj, err := jsonpatch.MergePatch(serializedObject, result.JsonPatch)
+		patch, err := jsonpatch.DecodePatch(result.JsonPatch)
+		if err != nil {
+			contextLogger.Error(err, "Error while decoding JSON patch from plugin", "patch", result.JsonPatch)
+			return nil, err
+		}
+
+		responseObj, err := patch.Apply(serializedObject)
 		if err != nil {
 			contextLogger.Error(err, "Error while applying JSON patch from plugin", "patch", result.JsonPatch)
 			return nil, err
